modules/storage/storage-account/main.bicep

```diff
@@ -0,0 +1,68 @@
+@description('Deployment Location')
+param location string
+
+@description('Name of Storage Account. Must be unique within Azure.')
+param name string = 'st${uniqueString(resourceGroup().id, subscription().id)}'
+
+@allowed([ 'new', 'existing' ])
+@description('Specifies whether to create a new Storage Account or use an existing one.')
+param newOrExisting string = 'new'
+
+@description('Name of Resource Group where the Storage Account is deployed when using an existing resource.')
+param existingResourceGroupName string = resourceGroup().name
+
+@description('ID of the subnet where the Storage Account will be deployed, if virtual network access is enabled.')
+param subnetID string = ''
+
+@description('Toggle to enable or disable virtual network access for the Storage Account.')
+param enableVNET bool = false
```

Comment:
Additionally would also be great if the module supported Private Endpoints (and what I was looking for with #235) in line with the approach in the other modules, for example: https://github.com/Azure/bicep-registry-modules/blob/main/modules/compute/container-registry/main.bicep#L258

Reply:
As per discussion below with @dciborow, we can follow up with another PR for this functionality.